website/static/js/apiPersonalToken.js

```diff
@@ -0,0 +1,419 @@
+                    name: this.name(),
+                    scopes: this.scopes().toString().replace(/,/g, ' ') || '',
+                    user_id: this.owner
+                }
+            }
+        };
+    }
+});
+
+/*
+ * Fetch data about tokens
+ */
+var TokenDataClient = oop.defclass({
+    /*
+     * Create the client for server operations on TokenData objects.
+     * @param {Object} apiListUrl: The api URL for token listing/creation
+     */
+    constructor: function (apiListUrl) {
+        this.apiListUrl = apiListUrl;
+    },
+    _fetchData: function (url) {
+        var ret = $.Deferred();
+        var request = $osf.ajaxJSON('GET', url, {isCors: true});
+
+        request.done(function (data) {
+            ret.resolve(this.unserialize(data));
+        }.bind(this));
+
+        request.fail(function (xhr, status, error) {
+            ret.reject(xhr, status, error);
+        }.bind(this));
+
+        return ret.promise();
+    },
+    fetchList: function () {
+        return this._fetchData(this.apiListUrl);
+    },
+    fetchOne: function (url) {
+        return this._fetchData(url);
+    },
+    _sendData: function (tokenData, url, method) {
+        var ret = $.Deferred();
+
+        var payload = tokenData.serialize();
+        var request = $osf.ajaxJSON(method, url, {isCors: true, data: payload});
+
+        request.done(function (data) { // The server response will contain the newly created/updated record
+            ret.resolve(this.unserialize(data));
+        }.bind(this));
+        request.fail(function (xhr, status, error) {
+            ret.reject(xhr, status, error);
+        });
+        return ret.promise();
+    },
+    createOne: function (tokenData) {
+        var url = this.apiListUrl;
+        return this._sendData(tokenData, url, 'POST');
+    },
+    updateOne: function (tokenData) {
+        var url = tokenData.apiDetailUrl;
+        return this._sendData(tokenData, url, 'PATCH');
+    },
+    deleteOne: function (tokenData) {
+        var url = tokenData.apiDetailUrl;
+        return $osf.ajaxJSON('DELETE', url, {isCors: true});
+    },
+    unserialize: function (apiData) {
+        var result;
+        // Check return type: return one object (detail view) or list of objects (list view) as appropriate.
+        if (Array.isArray(apiData.data)) {
+            result = $.map(apiData.data, function (item) {
+                return new TokenData(item);
+            });
+        } else if (apiData.data) {
+            result = new TokenData(apiData.data);
+        } else {
+            result = null;
+        }
+        return result;
+    }
+});
+
+/*
+  ViewModel for List views
+ */
+var TokensListViewModel = oop.defclass({
+    constructor: function (urls) {
+        this.apiListUrl = urls.apiListUrl;
+        this.webCreateUrl = urls.webCreateUrl;
+        // Set up data storage
+        this.tokenData = ko.observableArray();
+        this.sortedByName = ko.pureComputed(function () {
```

Comment:
I didn't know that was an API feature now- cool! Might make sense for a followup ticket (along with pagination)- an improvement, but not part of the scope of this ticket.